.github/workflows/ci.yml

```
@@ -0,0 +1,459 @@
+name: CI
+on:
+  workflow_call:
+    inputs:
+      # extra jobs must be multi-line string, as there's no support for `type: array` for inputs
+      extra_jobs:
+        type: string
+        required: false
+        default: ''
+      composer_require_extra:
+        type: string
+        required: false
+        default: ''
+      dynamic_matrix:
+        type: boolean
+        default: true
+      simple_matrix:
+        type: boolean
+        default: false
+      endtoend:
+        type: boolean
+        default: true
+      phpcoverage:
+        type: boolean
+        default: false
+      phplinting:
+        type: boolean
+        default: true
+      phpunit:
+        type: boolean
+        default: true
+      js:
+        type: boolean
+        default: true
+
+jobs:
+
+  context:
+    name: Context
+    runs-on: ubuntu-latest
+    env:
+      # Put this through an env variable to prevent possible string substitution injection via pull-request branch name
+      GITHUB_HEAD_REF: ${{ github.head_ref }}
```

```suggestion
      GITHUB_BASE_REF: ${{ github.base_ref }}
      GITHUB_HEAD_REF: ${{ github.head_ref }}
      GITHUB_REF: ${{ github.ref }}
      GITHUB_REF_NAME: ${{ github.ref_name }}
      GITHUB_REPOSITORY: ${{ github.repository }}
```
While it is extremely unlikely that these will have be usable for injection attacks, we should follow the best practice of always passing strings through this way.

You'll need to update the references in the step as well.

---

Updated except for github.repository, that's just a bit far fetched :-)  I've already tested elsewhere that github will sanatise the repo name on creation

github.base_ref is the target of any pull-request i.e. OUR branch, so it's not really exploitable.  However I've included as an env variable for less confusion - ref's get the env treatment

---

I went ahead and did all the github.repository on all the things.  I haven't for other github.* things like .token and .sha because there's no way for users to ever set those